examples_generator/libgen.py

```diff
@@ -1480,37 +1480,42 @@ def __init__(self, infra):
             infra.lengths[oget_track(otrack)] for otrack in range(2 * infra.nb_tracks)
         ]
 
-    def route_path(self, departure_utrack, arrival_utrack):
-        s1, s2 = uget_begin(departure_utrack), uget_end(departure_utrack)
-        t1, t2 = uget_begin(arrival_utrack), uget_end(arrival_utrack)
-        parent = [None for u in self.neighboors]
-        q = [(0, s1, -1), (0, s2, -1)]
-        while q != [] and parent[t1] == None and parent[t2] == None:
-            d, u, p = heappop(q)
-            if parent[u] == None:
-                parent[u] = p
-                for v in self.neighboors[u]:
-                    heappush(q, (d + self.weight[v], v, u))
-        path = []
-        u = t1 if parent[t1] != None else t2
-        while u != -1:
-            path.append(u)
-            u = parent[u]
-        path.reverse()
-        path = [oget_other_side(path[0])] + path + [oget_other_side(path[-1])]
-        return [oname_route_between(path[i], path[i + 1]) for i in range(len(path) - 1)]
-
-    def add_schedule(self, departure_time, departure_utrack, arrival_utrack):
-        assert departure_utrack != arrival_utrack
-        path = self.route_path(departure_utrack, arrival_utrack)
+    def route_path(self, track_list):
```

Comment:
This function seems to work. You can improve your code by spiting this function in two.
- One that compute a path between two given points (like before)
- One that compute a path between multiple points and using the first function.